core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/mod.rs

```diff
@@ -19,6 +21,54 @@ pub mod namespaces;
 #[cfg(test)]
 pub(crate) mod testonly;
 
+type RawParams<'a> = Option<Cow<'a, serde_json::value::RawValue>>;
+
+/// Version of `RawParams` (i.e., params of a JSON-RPC request) that is statically known to provide a borrow
+/// to another `RawParams` with the same lifetime.
+///
+/// # Why?
+///
+/// We need all this complexity because we'd like to access request params in a generic way without an overhead
+/// (i.e., cloning strings for each request – which would hit performance and fragment the heap).
+/// One could think that `jsonrpsee` should borrow params by default;
+/// if that were the case, we could opportunistically expect `RawParams` to be `None` or `Some(Cow::Borrowed(_))`
+/// and just copy `&RawValue` in the latter case. In practice, `RawParams` are *never* `Some(Cow::Borrowed(_))`
+/// because of a bug (?) with deserializing `Cow<'_, RawValue>`: https://github.com/serde-rs/json/issues/1076
```

Comment:
Here's motivation behind this design. I guess we could try fixing the bug (?) with `Cow<'_, RawValue>`, but I'm not sure where to even start with this: is it really a `serde_json` / `serde` bug? Or would it make sense to somehow fix deserialization in `jsonrpsee` (e.g., with manually implemented deserialization logic)?